pysat/ssnl/avg.py

```diff
@@ -46,8 +46,15 @@ def median2D(const, bin1, label1, bin2, label2, data_label,
 
     # create bins
     #// seems to create the boundaries used for sorting into bins
-    binx = np.linspace(bin1[0], bin1[1], bin1[2]+1)
-    biny = np.linspace(bin2[0], bin2[1], bin2[2]+1)
+    if range1 is not None:
```

Comment:
Stylistically, it would be cleaner to construct the if/then using 

`if range1 is None`